inlang/source-code/paraglide/paraglide-js/src/vite-plugin/index.ts

```diff
@@ -0,0 +1,82 @@
+import type { PluginOption } from "vite"
+import { exec } from "node:child_process"
+import { readFile } from "node:fs/promises"
+import type { ProjectSettings } from "@inlang/sdk"
+import { resolve } from "node:path"
+
+let cachedSettings: ProjectSettings | undefined = undefined
+
+export const paraglideJsVitePlugin = (config: {
+	settingsPath: string
+	namespace: string
+	silent?: boolean
+	timeout?: number
+	onInit?: boolean
+}): PluginOption => {
+	const options = {
+		silent: false,
+		timeout: 500,
+		onInit: true,
+		...config,
+	}
+
+	let throttled = false
+
+	const execute = () => {
+		exec(
+			`paraglide-js compile --namespace ${options.namespace} --project ${options.settingsPath}`,
+			(exception, output, error) => {
+				// eslint-disable-next-line no-console
+				if (!options.silent && output) console.log(output)
+				if (!options.silent && error) console.error(error)
+			}
+		)
+	}
+
+	return {
+		name: "vite-plugin-paraglide-js-watcher",
+
+		async buildStart() {
+			if (!cachedSettings) {
+				const settingsContent = await readFile(options.settingsPath, "utf-8")
+				cachedSettings = JSON.parse(settingsContent)
+			}
+
+			if (options.onInit) {
+				execute()
+			}
+		},
+
+		async handleHotUpdate({ file }) {
+			if (throttled) return
+
+			throttled = true
+
+			setTimeout(() => (throttled = false), options.timeout)
+
+			let filePath =
+				cachedSettings && (cachedSettings["plugin.inlang.messageFormat"]?.filePath as string)
```

Comment:
let's wait for file watcher before implementing hard coded stuff here which will break

Reply:
true. but overall concept is already established.